cachi2/core/package_managers/npm.py

```diff
@@ -476,6 +502,104 @@ def _get_npm_dependencies(
     return download_paths
 
 
+def get_type_of_dependency(dependency_version: str) -> Literal["registry", "git", "file", "https"]:
+    """Return type of dependency.
+
+    :param dependency_version: Dependency version loaded from package(-lock).json dependencies
+    """
+    url = urlparse(dependency_version)  # TODO: Do we need to support also http: ? I believe not
+    if url.scheme == "file":
+        return "file"
+    elif (
+        url.scheme == "git"
+        or url.scheme.startswith("git+")
+        or (url.scheme == "https" and url.path.endswith(".git"))
+    ):
+        return "git"
+    elif url.scheme == "https" and url.path.endswith(".tar.gz") or url.path.endswith(".tgz"):
+        return "https"
+    elif url.path.count("/") == 1:
+        # GitHub URLs are also supported in form of namespace/project e.g. "expressjs/express"
+        return "git"
+    else:
+        # Anything else should be considered as registry version or range of registry versions or latest
+        return "registry"
+
+
+def update_package_lock_with_local_paths(
+    download_paths: Dict[NormalizedUrl, RootedPath],
+    package_lock_file: ProjectFile,
+    output_dir: RootedPath,
+) -> ProjectFile:
+    """Replace packages resolved URLs with local paths.
+
+    Update package-lock.json file in a way it can be used in isolated environment (container)
+    without internet connection. All package/dependency URLs will be replaced for
+    local paths to downloaded dependencies.
+
+    :param download_paths:
+    :param package_lock_file: File path to package-lock.json
+    :param output_dir: Output directory with fetched dependencies
+    """
+    package_lock = PackageLock.from_file(RootedPath(package_lock_file.abspath))
+
+    if package_lock.lockfile_version > 1:
+        for package in package_lock._get_packages():
+            # Remove integrity for git sources, their integrity checksum will change when
+            # constructing tar archive from cloned repository
+            url = _normalize_resolved_url(package.resolved_url)
+            if _classify_resolved_url(url) == "git":
+                if package.integrity:
+                    package.integrity = ""
+
+            # Update all type of dependencies except file:
+            if _classify_resolved_url(url) != "file":
+                templated_abspath = str(download_paths[url].path).replace(
+                    str(output_dir.path), "${output_dir}"
+                )
+                package.resolved_url = f"file://{templated_abspath}"
+
+            if package.dependencies:
+                for dependency, dependency_version in package.dependencies.items():
```

Comment:
Don't forget the other types of dependencies :slightly_smiling_face: 

https://github.com/containerbuildsystem/cachito/blob/6fa22becb705c42c47f3fb01fdb7dc21f49c5ff7/cachito/workers/pkg_managers/npm.py#L184

Reply:
Added, thanks for reminding me.